tools/gnur-runtime/src/main/java/org/renjin/gnur/Sort.java

```diff
@@ -33,24 +36,53 @@
    * larger to right
    *
    */
-  public static void rPsort2(DoublePtr x, int lo, int hi, int k) {
+  public static void iPsort2(Ptr x, int lo, int hi, int k) {
+    boolean nalast = true;
+    int v, w;
+    int L, R, i, j;
+
+    for (L = lo, R = hi; L < R;) {
+      v = x.getInt(k);
+      for (i = L, j = R; i <= j;) {
+        while (icmp(x.getInt(i), v, nalast) < 0) {
```

Comment:
I think in nearly all cases, `getInt()` should be `getAlignedInt()` and `getDouble()` should be `getAlignedDouble()`. All of the `getXXX()` methods accept an argument as the number of _bytes_ from the beginning of the pointer, while the `getAlignedXXX()` methods accept an argument as the number of elements.